lib/praxis/resource_definition.rb

```diff
@@ -11,7 +11,12 @@ module ResourceDefinition
       @version = 'n/a'.freeze
       @actions = Hash.new
       @responses = Hash.new
-      @action_defaults = Trait.new
+
+      # Ensure we inherit any base params defined in the API definition for this version
+      base_attrs = if base_params = ApiDefinition.instance.info.base_params
```

Comment:
not really, since we need to calculate the defaults based on different version parts of it.
What I can do, though is to pass in the version instead, and calculate the attributes inside the generate_defaults_block, which I agree it is cleaner.